bundles/org.openhab.binding.chatgpt/src/main/java/org/openhab/binding/chatgpt/internal/ChatGPTHandler.java

```diff
@@ -79,59 +77,106 @@ public ChatGPTHandler(Thing thing, HttpClientFactory httpClientFactory) {
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
-        if (command instanceof RefreshType && !"".equals(lastPrompt)) {
-            String response = sendPrompt(channelUID, lastPrompt);
-            processChatResponse(channelUID, response);
-        }
 
         if (command instanceof StringType stringCommand) {
             lastPrompt = stringCommand.toFullString();
-            String response = sendPrompt(channelUID, lastPrompt);
-            processChatResponse(channelUID, response);
+
+            String queryJson = prepareRequestBody(channelUID);
+
+            if (queryJson != null) {
+                String response = sendPrompt(queryJson);
+                processChatResponse(channelUID, response);
+            }
         }
     }
 
     private void processChatResponse(ChannelUID channelUID, @Nullable String response) {
         if (response != null) {
-            ChatResponse chatResponse = gson.fromJson(response, ChatResponse.class);
+
+            ObjectMapper objectMapper = new ObjectMapper();
+            ChatResponse chatResponse;
+            try {
+                chatResponse = objectMapper.readValue(response, ChatResponse.class);
+            } catch (JsonProcessingException e) {
+                logger.error("Failed to parse ChatGPT response: {}", e.getMessage(), e);
```

Comment:
I assume this is really an unexpected error condition where users should file a bug ticket for, if it occurs?